### PR TITLE
publish/deploy via CI

### PR DESCRIPTION
**Pull request by TBonnin**
## Describe your changes
This PR allows to build/deploy Nango services and publish npm packages via CI. No more building locally anymore.

I will write some doc once this is merged and I have the proper link for the publish/deploy workflow

To publish npm packages:
1. Go to Github Actions https://github.com/NangoHQ/nango/actions
2. Select the `[Release] Npm Publish` workflow
3. Click on `Run worklfow` (right side)
4. Enter version (ex: 0.36.77) and submit

The workflow will publish the node-client, shared, CLI and frontend packages if the version doesn't already exist. It will also update the package.json files and commit to master

To deploy a nango servicer
1. Wait for the `[Release] Build images` to complete. It runs on every push.
2. Select the `[Release] Deploy service` worflow
3. Click on `Run workflow`
4. Select services (server, jobs or runner) and the stage (production or staging)

## Issue ticket number and link
https://linear.app/nango/issue/NAN-211/release-via-ci

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: .github/workflows/release/build-images.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            package: runner
+            run-cmd: ts-build
+            tags: -t nangohq/nango-runner:${{ github.sha }}
+    nango-server-staging:
+        uses: ./.github/workflows/release/push-container.yaml
+        secrets: inherit
+        with:
+            package: server
+            run-cmd: build:staging
+            tags: -t nangohq/nango-server:staging-${{ github.sha }}
```
Comment (TBonnin) [inline .github/workflows/release/build-images.yaml:27]: I propose to use a single docker repo (`nangohq/nango-server`) for all server images with 4 tag tracks:
- production
- staging
- hosted
- enterprise

If you are ok with it. I will then modify the `nango-render-cloud` repo to use the images for prod and staging

File: packages/cli/package.json [hunk 2]
```diff
@@ -4,21 +4,23 @@
...
+        "nango": "dist/index.js"
+    },
     "types": "dist/nango-sync.d.ts",
     "keywords": [],
     "author": "bastien@nango.dev",
     "repository": {
         "type": "git",
-        "url": "https://github.com/NangoHQ/nango",
+        "url": "git+https://github.com/NangoHQ/nango.git",
         "directory": "packages/cli"
```
Comment (TBonnin) [inline packages/cli/package.json:16]: unrelated to this PR. I just ran `npm pkg fix` to remove the warning when publishing

File: packages/cli/package.json [hunk 3]
```diff
@@ -60,6 +59,9 @@
         "@types/npm-package-arg": "^6.1.1",
         "@types/promptly": "^3.0.2",
         "babel-loader": "^9.1.2",
+        "esbuild": "^0.17.19",
+        "ts-node": "^10.9.1",
+        "typescript": "^5.0.4",
```
Comment (TBonnin) [inline packages/cli/package.json:64]: unrelated as well. moving compiler and builder out of the dependencies to the devDep

**Review by khaliqgant — APPROVED** [review]
Looks good. Just curious what `stefanzweifel/git-auto-commit-action@v5` looks like and to verify that it triggers the https://github.com/NangoHQ/nango/blob/master/.github/workflows/integration-flows.yaml workflow

**Comment by TBonnin** [comment]
> Looks good. Just curious what `stefanzweifel/git-auto-commit-action@v5` looks like and to verify that it triggers the https://github.com/NangoHQ/nango/blob/master/.github/workflows/integration-flows.yaml workflow

It commits all changes made by the action. Because we are using a admin access token to bypass the fact that master is a protected branch. This commit actually triggers all actions, even the publish action, which would not publish anything since the version number would match what's in npm but it is a bit of a waste

**Pull request merged**